tools/query_verification/src/main/java/com/google/bigquery/TeradataManager.java

```diff
@@ -0,0 +1,364 @@
+package com.google.bigquery;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.sql.*;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Class to communicate with TD to create tables and run queries using the JDBC driver.
+ */
+public class TeradataManager implements DataWarehouseManager {
+
+    private final QueryVerificationQuery query;
+    private final QueryVerificationSchema schema;
+    private final List<QueryVerificationData> data;
+
+    private Statement statementConnection;
+    private Connection csvConnection;
+
+    public TeradataManager(QueryVerificationQuery query, QueryVerificationSchema schema, List<QueryVerificationData> data) {
+        this.query = query;
+        this.schema = schema;
+        this.data = data;
+    }
+
+    @Override
+    public String getName() {
+        return "Teradata";
+    }
+
+    /**
+     * Sends query jobs to TD
+     * @return List of query results
+     * @throws Exception
+     */
+    @Override
+    public List<QueryJobResults> runQueries() throws Exception {
+        setupConnection();
+
+        List<String> tables = createTablesFromSchema();
+        populateTablesFromData();
+
+        // Create query jobs
+        List<String> statements = getStatementsFromQuery();
+
+        // Store results from every job
+        List<QueryJobResults> jobResults = new ArrayList<QueryJobResults>();
+
+        for (int i = 0; i < statements.size(); i++) {
+            String statement = statements.get(i);
+
+            QueryJobResults jobResult;
+            try {
+                // Run query job
+                ResultSet resultSet = statementConnection.executeQuery(statement);
+
+                // Parse and store query results
+                List<List<String>> rawResults = new ArrayList<List<String>>();
+                Set<List<Object>> results = new HashSet<List<Object>>();
+                ResultSetMetaData resultSetMetaData = resultSet.getMetaData();
+
+                while (resultSet.next()) {
+                    List<String> rowRawResults = new ArrayList<String>();
+                    for (int j = 1; j <= resultSetMetaData.getColumnCount(); j++) {
```

Comment:
Maybe have some comments to explain why not starting with 0

Reply:
Updated. The reason was that `ResultSet` column indexes start at 1 instead of 0